### PR TITLE
Refactor/329 RSR 681 refactor and restruct extrefservice, fix #329

### LINKED ISSUE #329: [RSR-681] - organizing ExtRefService
Restructuring ExtRefService to separate functions related to Extref and ControlBlock.

### PR DESCRIPTION
**Pull request by samirromdhani**
Fix https://github.com/com-pas/compas-sct/issues/329

### CODE REVIEW

**Comment by GuillaumeJAFFRE** [comment]
Il ne semble pas y avoir la même norme de nommage pour les services / editor / test sur Scl / ControlBlock / ExtRef

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+import java.util.*;
+import java.util.stream.Stream;
+
+public class ControlBlockService implements ControlBlockEditor {
+
+
+    @Override
+    public List<SclReportItem> analyzeDataGroups(SCL scd) {
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
```
Comment (gleizesDor) [inline sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java:28]: Is it the good time to get rid of these "adapter" ?

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                    list.addAll(iedAdapter.checkBindingDataGroupCoherence());
+                    return list;
+                }).flatMap(Collection::stream).toList();
+    }
+
+    @Override
+    public List<SclReportItem> createDataSetAndControlBlocks(SCL scd, Set<FcdaForDataSetsCreation> allowedFcdas) {
+        checkFcdaInitDataPresence(allowedFcdas);
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
+        Stream<LDeviceAdapter> lDeviceAdapters = sclRootAdapter.streamIEDAdapters().flatMap(IEDAdapter::streamLDeviceAdapters);
```
Comment (gleizesDor) [inline sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java:42]: Is it the good time to get rid of these "adapter" ?

```java
Stream<TLDevice> tlDeviceStream = scd.getIED()
                .stream()
                .filter(TIED::isSetAccessPoint)
                .flatMap(tied -> tied.getAccessPoint().stream())
                .map(TAccessPoint::getServer)
                .filter(Objects::nonNull)
                .filter(TServer::isSetLDevice)
                .flatMap(tServer -> tServer.getLDevice().stream());
```

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        return createDataSetAndControlBlocks(lDeviceAdapters, allowedFcdas);
+    }
+
+    @Override
+    public List<SclReportItem> createDataSetAndControlBlocks(SCL scd, String targetIedName, Set<FcdaForDataSetsCreation> allowedFcdas) {
+        checkFcdaInitDataPresence(allowedFcdas);
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
```
Comment (gleizesDor) [inline sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java:49]: Is it the good time to get rid of these "adapter" ?
EveryWhere

Comment (massifben) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java:49]: image

Comment (samirromdhani) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java:49]: It's so funny picture.

Comment (samirromdhani) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java:49]: I just include the refactoring that was specified.. https://gopro-tickets.rte-france.com/browse/RSR-681

Comment (samirromdhani) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java:49]: Hello @gleizesDor, @massifben ,
I think there are specific tickets for dealing with adapters.

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java (hunk 1 above)
Comment (gleizesDor) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java:28]: ok next time

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java (hunk 3 above)
Comment (gleizesDor) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java:49]: ok next time

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java (hunk 2 above)
Comment (gleizesDor) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java:42]: ok next time

**Review by gleizesDor — APPROVED** [review]

**Pull request merged**